nikola/nikola.py

```diff
@@ -1334,9 +1337,15 @@ def render_template(self, template_name, output_name, context, url_type=None):
 
         utils.makedirs(os.path.dirname(output_name))
         parser = lxml.html.HTMLParser(remove_blank_text=True)
-        doc = lxml.html.document_fromstring(data, parser)
+        if is_fragment:
+            doc = lxml.html.fragment_fromstring(data, parser)
+        else:
+            doc = lxml.html.document_fromstring(data, parser)
         self.rewrite_links(doc, src, context['lang'], url_type)
-        data = b'<!DOCTYPE html>\n' + lxml.html.tostring(doc, encoding='utf8', method='html', pretty_print=True)
+        if is_fragment:
+            data = (doc.text or '').encode('utf-8') + ''.encode('utf-8').join([lxml.html.tostring(child, encoding='utf-8', method='html') for child in doc.iterchildren()])
```

Comment:
`doc.text` is all the text in `doc` that appears before a HTML element. Text that appears after HTML elements is taken care of by `tostring()`. (lxml is weird.)